tests/TestCase/Controller/ControllerTest.php

```diff
@@ -474,6 +474,22 @@ public function testRedirectByCode($code, $msg)
     }
 
     /**
+     * testRedirect method
+     *
+     * @return void
+     */
+    public function testRedirectSameUrl()
+    {
+        $Controller = new Controller(null);
+        $Controller->response = new Response();
+
+        $response = $Controller->redirect(null);
```

Comment:
I don't think the explicit null is necessarily bad or something we _need_ to fix, but I can see why allowing `redirect()` would be nice as it mirrors `Router::url()`.